metrics.nim

```diff
@@ -683,3 +687,125 @@ proc startHttpServer*(address = "127.0.0.1", port = Port(9093)) =
   when defined(metrics):
     httpServerThread.createThread(httpServer, (address, port))
 
+################
+# process info #
+################
+
+when defined(metrics) and defined(linux):
+  import posix
+
+  type ProcessInfo = ref object of Gauge
+
+  proc newProcessInfo*(name: string, help: string, registry = defaultRegistry): ProcessInfo =
+    validateName(name)
+    result = ProcessInfo(name: name,
+                        help: help,
+                        typ: "gauge", # Prometheus won't allow fantasy types in here
+                        creationThreadId: getThreadId())
+    result.register(registry)
+
+  var
+    processInfo* {.global.} = newProcessInfo("process_info", "CPU and memory usage")
```

Comment:
Why are these global variables special? Shouldn't these be fields in an object that I decide to instantiate myself?

Reply:
They should register by default and allow users to unregister them, according to the Prometheus docs.

They're {.global.} because I wanted to make sure they're only created (and registered) once.